api/app/signals/apps/signals/models/area.py

```diff
@@ -0,0 +1,114 @@
+from django.apps import apps
+from django.contrib.gis.db import models
+from django.db import transaction
+from django.dispatch import Signal as DjangoSignal
+
+action_create_initial = DjangoSignal(providing_args=['model', 'instance'])
+action_update = DjangoSignal(providing_args=['model', 'instance', 'prev_instance'])
+
+
+class AreaType(models.Model):
+    name = models.CharField(max_length=255)
+    code = models.CharField(max_length=255)
+    description = models.TextField(max_length=3000)
+
+
+# We want to track the creation and edit history of an area. Since area
+# instances do not comprise a lot of data, we just copy the data to a history
+# table that shadows the Area table.
+
+
+class HistoryManager(models.Manager):
+    """Base class for ModelManagers that track history for a model"""
+    TRACKED_MODEL = None  # set to correct value in sub class
+    HISTORY_MODEL = None  # set to correct value in sub class
+
+    def create_initial(self, user, **data):
+        tracked_model = apps.get_model(self.TRACKED_MODEL)
+        history_model = apps.get_model(self.HISTORY_MODEL)
+
+        with transaction.atomic():
+            new_instance = tracked_model.objects.create(**data)
+
+            history_data = data.copy()
+            history_data['_tracked_pk'] = new_instance.id
```

Comment:
The primary key can be accessed by using the pk in the Django ORM, use that instead of the id